### ISSUE TITLE
Add slots for pagination

### ISSUE THREAD

**Issue by luohy15** [carbon-design-system/carbon-components-vue#453]
**Is your feature request related to a problem? Please describe.**
I want to customize the rangeText and pageOfPages of cv-pagination like this:
![](https://user-images.githubusercontent.com/19179374/59824074-b4265580-9362-11e9-97d9-cef16deecfde.png)

**Describe the solution you'd like**
Use slots.

**Describe alternatives you've considered**
Add named scoped slots with fallback content like this:
`<slot name='range' :number="numberOfItems">{{ rangeText }}</slot>`
`<slot name='page' :page="pageCount">{{ pageOfPages }}</slot>`

**Additional context**
Need to consider which variables are exposed to the user.
cv-data-table also need to be modified .

**Comment by lee-chase** [comment]
@luohy15 would #463 using scoped slots work for you?

**Comment by luohy15** [comment]
Yes, it works for me! Thanks a lot! Now I can use
```
<template slot="range-text" slot-scope="{scope}">{{`共${scope.items}条`}}</template>
<template slot="of-n-pages" slot-scope="{scope}">{{`共${scope.pages}页`}}</template>
```
to achieve my goal.
Actually we also often using cv-pagination inside cv-data-table, so I'm thinking that we can [Pass Slots through](https://gist.github.com/Loilo/73c55ed04917ecf5d682ec70a2a1b8e2) from CvDataTable to CvPagination like this:
```
<template slot="range-text" slot-scope="{scope}"><slot name="range-text" v-bind:scope="scope">{{toRangeText(scope)}}</slot></template>
<template slot="of-n-pages" slot-scope="{scope}"><slot name="of-n-pages" v-bind:scope="scope">{{toPageOfPages(scope)}}</slot></template>
```
and use the same toRangeText/toPageOfPages function in both cv-pagination and cv-data-table.

**Comment by lee-chase** [comment]
Published today in 2.7.0
